### PR TITLE
feat(gatsby-plugin-typescript): Add schema 

### PR DESCRIPTION
**Pull request by laurieontech**
This makes use of the new plugin schema for `gatsby-plugin-typescript`

### CODE REVIEW

File: packages/gatsby-plugin-typescript/src/gatsby-node.js [hunk 1]
```diff
@@ -35,6 +35,18 @@ function onCreateWebpackConfig({ actions, loaders }) {
...
 }
 
+exports.pluginOptionsSchema = ({ Joi }) => {
+  Joi.object({
+    isTSX: Joi.boolean().description(`Enables jsx parsing.`).default(true),
+    jsxPragma: Joi.string()
+      .description(`Replace the function used when compiling JSX expressions.`)
+      .default(`React`),
+    allExtensions: Joi.boolean()
+      .description(`Indicates that every file shoudl be parsed as TS or TSX.`)
```
Comment (ascorbic) [inline packages/gatsby-plugin-typescript/src/gatsby-node.js:45]:
```suggestion
      .description(`Indicates that every file should be parsed as TS or TSX.`)
```

**Review by mfrachet — COMMENTED** [review]
Thank you for taking care of this one 🙏 

Could you please validate the schema using https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-utils/src/test-plugin-options-schema.ts ?

There's an example in Netlify's plugin (https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-netlify)

I think it could help people having references on how to test their own schemas 😊

File: packages/gatsby-plugin-typescript/src/gatsby-node.js [hunk 2]
```diff
@@ -35,6 +35,21 @@ function onCreateWebpackConfig({ actions, loaders }) {
   })
 }
 
+if (process.env.GATSBY_EXPERIMENTAL_PLUGIN_OPTION_VALIDATION) {
+  exports.pluginOptionsSchema = ({ Joi }) =>
+    Joi.object({
+      isTSX: Joi.boolean().description(`Enables jsx parsing.`).default(true),
```
[the remark this answers is not in the document]
Comment (mxstbr) [reply packages/gatsby-plugin-typescript/src/gatsby-node.js:41]: We're making it work, so please add defaults! :+1:

**Review by mfrachet — APPROVED** [review]
🙏 thank you so much!

**Pull request merged**